src/third_party/starboard/wpe/shared/player/player_internal.cc

```diff
@@ -457,6 +457,12 @@ enum class MediaType {
 
 constexpr char kClearSamplesKey[] = "fake-key-magic";
 
+#if SB_HAS(NATIVE_AUDIO)
+static constexpr GstStreamVolumeFormat volumeFormat = GST_STREAM_VOLUME_FORMAT_LINEAR;
+#else
+static constexpr GstStreamVolumeFormat volumeFormat = GST_STREAM_VOLUME_FORMAT_CUBIC;
+#endif
+
```

Comment:
I do not think making it dependent on NATIVE_AUDIO is the way to go since this effectively changes volume format for platforms which does not define that (which is all currently AFAICS). This might be ok but then let's do it explicit and switch to CUBIC format for all but then passing 0-1 values as cubic volume does not look ok to me.

Reply:
In the validation tools of gstreamer including gst-play and launch, by default it uses Cubic and values as far as I can see between 0-1. https://github.com/GStreamer/gst-plugins-base/blob/master/tools/gst-play.c#L309

Reply:
This isn't that apparent from me from the code you mentioned because it:
1. Takes the volume in cubic
2. clamp it to 0-10

Reply:
Inner calculation can be different, same goes for Webkit, but the point is higher api is still getting a value between 0..1. In gst-play each volume up arrow gets with volume_step of 1/20 = 0.05 which means %5 per step. 

We need to find a common way for both webkit and cobalt. In case of both apps are running on same platform, webkit will send Linear when there is native-audio otherwise cubic. Cobalt sends always Linear. As the implementation of audiosink is single, it can not handle these behaviours.

Reply:
BTW, even webkit has different behaviours based on different versions:

wpe-2.22 Always CUBIC - https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/wpe-2.22/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp#L661 

wpe-2017 Native-Audio LINEAR, otherwise CUBIC: https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/wpe-20170728/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp#L562

webkit upstream: Always LINEAR: https://github.com/WebKit/webkit/blob/master/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L1755

Reply:
The thing is that sink should implement "volume" property in linear. GST interprets the the passed in value in format you say it is and converts to to linear. I think YT sets in linear so when setting 0.5 you should be having half of the full range of volume where with CUBIC you'll end up with 0.125